### PR TITLE
[MNOE-1285] date format by locale

### PR DESCRIPTION
**Pull request by iseessel**

### CODE REVIEW

File: src/app/services/locale-helper/locale-config.svc.coffee [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  # Find the locale from the Tenant#settings
+  localeFromTenant = ->
+    MnoeTenant.get().then((response) ->
+      response.data?.frontend_config?.system?.i18n?.preferred_locale
+    )
+
+  # Build our fallback stack manually to be ['language', preferredLanguage, LOCALES.fallbackLanguage]
+  # eg: If the detected locale is 'fr-FR' and the preferred language 'en-GB', the fallback stack is
+  # ['fr', 'en-GB', 'en']
+  # This is similar to the angular-translate implementation except they push th preferredLanguage at the end
```
Comment (alexmaestrano) [inline src/app/services/locale-helper/locale-config.svc.coffee]: 👍

File: src/app/views/organization/app-instance-info-modal/app-instance-info-modal.html [hunk 2]
```diff
@@ -93,7 +93,7 @@ <h3 class="modal-title" translate translate-values="{ app_name: vm.appInstance.a
         <!-- TODO: Figure out what time zone to put this in. Time zone of customer? -->
         <div class="col-xs-6">
           <span translate>
-            {{ vm.syncStatus.started_at | date: "MM/dd/yyyy 'at' h:mma" | notApplicable }}
+            {{ vm.syncStatus.started_at | amDateFormat:"L [at] h:mma | notApplicable"
```
Comment (alexmaestrano) [inline src/app/views/organization/app-instance-info-modal/app-instance-info-modal.html]: Quote typo / missing curly brace?

File: src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html [hunk 3]
```diff
@@ -22,8 +22,8 @@
         </tbody>
         <tbody ng-show="!$ctrl.events.loading">
           <tr ng-repeat="event in $ctrl.events.list">
-            <td title="{{ event.created_at | amDateFormat: 'DD/MM/YYYY, h:mm:ss a' }}">
-              {{event.created_at | amCalendar }}
+            <td title="{{ event.created_at | amDateFormat: 'L h:mm:ss a' }}">
```
Comment (ouranos) [inline src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html]: Change to `'L LTS'`, for country using 24 hours clock (eg France) we'll get: `28/10/2018 16:25:21`
For country using 12  hours clock we'll get: `10/28/2018 4:25:21 PM`

File: src/app/components/mnoe-dashboards-list/mnoe-dashboards-list.html [hunk 4]
```diff
@@ -26,8 +26,8 @@
               {{ dashboard.name }}
             </td>
             <td title="{{ dashboard.full_name }}">{{ $ctrl.currency(dashboard) }}</td>
-            <td title="{{ dashboard.full_name }}">{{ dashboard.updated_at | date: "dd/MM/yyyy \'at\' h:mm a" }}</td>
-            <td title="{{ dashboard.full_name }}">{{ dashboard.created_at |  date: "dd/MM/yyyy \'at\' h:mm a" }}</td>
+            <td title="{{ dashboard.full_name }}">{{ dashboard.updated_at | amDateFormat:"L [at] h:mm a" }}</td>
```
Comment (ouranos) [inline src/app/components/mnoe-dashboards-list/mnoe-dashboards-list.html]: We should get rid of the `[at]` as it's not translated, just escaped.
"L LT" should be fine?

File: src/app/components/mnoe-invoices-list/mnoe-invoices-list.html [hunk 5]
```diff
@@ -22,8 +22,8 @@
           <tr ng-repeat="invoice in $ctrl.invoices.list" ng-click="$ctrl.goToInvoice(invoice.id)">
             <td>{{invoice.slug}}</td>
             <td><a href="">{{invoice.organization.name}}</a></td>
-            <td>{{invoice.started_at | amCalendar}} - {{invoice.ended_at | amCalendar}}</td>
-            <td ng-if="$ctrl.isPaymentEnabled">{{$ctrl.expectedPaymentDate(invoice.ended_at) | amCalendar }}</td>
+            <td>{{invoice.started_at | amDateFormat:'L z'}} - {{invoice.ended_at | amDateFormat:'L z'}}</td>
```
Comment (ouranos) [inline src/app/components/mnoe-invoices-list/mnoe-invoices-list.html]: Keep "amCalendar"?

File: src/app/components/mnoe-users-list/mno-users-list.coffee [hunk 6]
```diff
@@ -52,7 +52,7 @@
           style: {width: "130px"}
           render: (user) ->
             template: """
-            <span data-toggle="tooltip" title="{{::user.last_sign_in_at | date: 'H:m - dd/MM/yyyy'}}">
+            <span data-toggle="tooltip" title="{{::user.last_sign_in_at | amDateFormat: 'H:m [-] L'}}">
```
Comment (ouranos) [inline src/app/components/mnoe-users-list/mno-users-list.coffee]: Let's use "L LT" for consistency?

File: src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html (hunk 3 above)
Comment (iseessel) [reply src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html]: 👍

File: src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html [hunk 7]
```diff
@@ -22,8 +22,8 @@
...
+              {{event.created_at | amDateFormat:'L z' }}
```
[the remark this answers is not in the document]
Comment (iseessel) [reply src/app/components/mnoe-audit-events-list/mnoe-audit-events-list.html]: Ah my bad -- I was copying logic from OPAL-568, which was done on opal-webstore -- but failed to realize the specs were different.

File: src/app/components/mnoe-dashboards-list/mnoe-dashboards-list.html (hunk 4 above)
Comment (iseessel) [reply src/app/components/mnoe-dashboards-list/mnoe-dashboards-list.html]: 👍 yeah agreed the `[at]` not great, will check if` L LT` works.

File: src/app/components/mnoe-invoices-list/mnoe-invoices-list.html (hunk 5 above)
Comment (iseessel) [reply src/app/components/mnoe-invoices-list/mnoe-invoices-list.html]: 👍

File: src/app/components/mnoe-users-list/mno-users-list.coffee (hunk 6 above)
Comment (iseessel) [reply src/app/components/mnoe-users-list/mno-users-list.coffee]: 👍

**Pull request merged**